src/chunkserver/datastore/chunkserver_chunkfile.cpp

```diff
@@ -189,7 +192,14 @@ CSErrorCode CSChunkFile::Open(bool createFile) {
         char buf[pageSize_];  // NOLINT
         memset(buf, 0, sizeof(buf));
         metaPage_.encode(buf);
-        int rc = chunkFilePool_->GetFile(chunkFilePath, buf);
+
+        auto updateMetapageCb = [this](char* buffer) {
```

Comment:
We can't gurantee get clean chunk success in every operation, so if we get dirty chunk we should use `FORMAT_VERSION `

Reply:
> We can't gurantee get clean chunk success in every operation, so if we get dirty chunk we should use `FORMAT_VERSION ` in metapage.

I think you will get a clean chunk if you set `needClean` to true when calling `GetChunk`.
And after `GetChunk` returned with true, you got a cleaned chunk even if the chunk is from dirty pool, because you did fallocate it with `FALLOC_FL_ZERO_RANGE`. And for `FALLOC_FL_ZERO_RANGE`, it will mark the file as unwritten, so read from this file is just like a clean chunk file.

Reply:
@wu-hanqing done